### ISSUE TITLE
New rule proposal: TSLint's no-default-export

### ISSUE THREAD

**Issue by gruckion** [eslint/eslint#11783]
**Please describe what the rule should do:**

Disallows default exports in ES6-style modules.

Use named exports instead.

https://palantir.github.io/tslint/rules/no-default-export/

**What category of rule is this? (place an "X" next to just one item)**

[X] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Disallow
```js
export default Bob = 5;
```

Allowed
```js
export const Bob = 5
```

**Why should this rule be included in ESLint (instead of a plugin)?**

Named imports/exports promote clarity. In addition, current tooling differs on the correct way to handle default imports/exports. Avoiding them all together can help avoid tooling bugs and conflicts.

**Are you willing to submit a pull request to implement this rule?**

Maybe

**Comment by platinumazure** [comment]
Hi @gruckion, thanks for the issue.

I think this can already be achieved using [`no-restricted-syntax`](https://eslint.org/docs/rules/no-restricted-syntax):

```js
/* eslint no-restricted-syntax: ["error", { "selector": "ExportDefaultDeclaration", message: "Use named exports" }] */

export default const foo = "bar";
// Will show a warning
```

**Comment by g-plane** [comment]
@gruckion You can also consider [eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import):
https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-default-export.md

**Comment by gruckion** [comment]
Thank you, will check out both suggestions.

**Comment by g-plane** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
